canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/RealTimeFrameProcessor.java

```diff
@@ -0,0 +1,86 @@
+package com.amazon.kinesis.video.canary.consumer;
+
+import com.amazonaws.kinesisvideo.parser.mkv.Frame;
+import com.amazonaws.kinesisvideo.parser.mkv.FrameProcessException;
+import com.amazonaws.kinesisvideo.parser.utilities.FragmentMetadata;
+import com.amazonaws.kinesisvideo.parser.utilities.FragmentMetadataVisitor;
+import com.amazonaws.kinesisvideo.parser.utilities.FrameVisitor;
+import com.amazonaws.kinesisvideo.parser.utilities.MkvTrackMetadata;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchAsync;
+import com.amazonaws.services.cloudwatch.model.Dimension;
+import com.amazonaws.services.cloudwatch.model.MetricDatum;
+import com.amazonaws.services.cloudwatch.model.PutMetricDataRequest;
+import com.amazonaws.services.cloudwatch.model.StandardUnit;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.CRC32;
+
+import org.apache.commons.io.FileUtils;
+
+import java.util.Date;
+
+import java.util.Scanner;
+import java.io.FileReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+
+public class RealTimeFrameProcessor extends WebrtcStorageCanaryConsumer implements FrameVisitor.FrameProcessor {
+
+    boolean isFirstFrame = false;
+
+    public static RealTimeFrameProcessor create() {
+        return new RealTimeFrameProcessor();
+    }
+
+    private RealTimeFrameProcessor() {
+    }
+
+    @Override
+    public void process(Frame frame, MkvTrackMetadata trackMetadata, Optional<FragmentMetadata> fragmentMetadata, Optional<FragmentMetadataVisitor.MkvTagProcessor> tagProcessor) throws FrameProcessException {
+        System.out.println("RealTimeFrameProcessor invoked for frame" + frame);
+        if (!isFirstFrame) {
```

Comment:
Should add a check to only process the metrics if the stream has been empty, otherwise the consumer end may have started too late to catch the true "first frame received" time